src/silx/gui/widgets/CollapsibleWidget.py

```diff
@@ -0,0 +1,47 @@
+from silx.gui import qt
+
+
+class CollapsibleWidget(qt.QWidget):
+    """A widget with an inner layout that can collapsed/expanded by clicking on a button
+
+    :param title: Title of the widget. Will be used as the button label.
+    """
+
+    def __init__(self, title: str = "", parent: qt.QWidget | None = None) -> None:
+        super().__init__(parent=parent)
+        layout = qt.QVBoxLayout(self)
+        layout.setContentsMargins(0, 0, 0, 0)
+
+        self._toolButton = qt.QToolButton()
+        self._toolButton.setSizePolicy(
+            qt.QSizePolicy.Policy.Expanding, qt.QSizePolicy.Policy.Fixed
+        )
+        self._toolButton.setText(title)
+        self._toolButton.setToolButtonStyle(
+            qt.Qt.ToolButtonStyle.ToolButtonTextBesideIcon
+        )
+        self._toolButton.setCheckable(True)
+        self._toolButton.toggled.connect(self._toggleIcon)
+        self._toolButton.setChecked(True)
+        layout.addWidget(self._toolButton)
+
+        self._contentsFrame = qt.QGroupBox()
+        layout.addWidget(self._contentsFrame)
+        self._toolButton.toggled.connect(self._contentsFrame.setVisible)
+
+    def setInnerLayout(self, layout: qt.QLayout | None) -> None:
+        """Sets the inner collapsible layout"""
```

Comment:
```suggestion
    def setContentsLayout(self, layout: qt.QLayout | None) -> None:
        """Sets the inner collapsible layout"""
```
?

To mimic `setContentsMargins` :shrug: , up to you.